backend/fms_core/utils.py

```diff
@@ -192,8 +192,13 @@ def make_timestamped_filename(file_name: str) -> str:
     return f"{name}_{time.strftime('%Y-%m-%d_%H-%M-%S')}{extension}"
 
 
-WarningType = NewType('WarningType', dict[str, tuple[str] | str | list[str] | list[tuple[str, list]]])
-def serialize_warnings(warnings: WarningType):
+WarningsType = NewType('WarningsType', dict[str, tuple[str] | str | list[str] | list[tuple[str, list]]])
+class SerializedWarningTypeDict(TypedDict):
```

Comment:
warningtype... warningstype ... i would recommend sticking to one this is kind of stuff that generate typos.